test/Yoti.Auth.Tests/ShareUrl/Extensions/LocationConstraintExtensionBuilderTests.cs

```diff
@@ -0,0 +1,100 @@
+        {
+            Assert.ThrowsException<ArgumentException>(() =>
+            {
+                new LocationConstraintExtensionBuilder()
+                .WithLatitude(latitude)
+                .Build();
+            });
+        }
+
+        [DataTestMethod]
+        [DataRow(-181)]
+        [DataRow(181)]
+        [TestMethod]
+        public void ShouldFailForLongitudesOutsideOfRange(double longitude)
+        {
+            Assert.ThrowsException<ArgumentException>(() =>
+            {
+                new LocationConstraintExtensionBuilder()
+                .WithLongitude(longitude)
+                .Build();
+            });
+        }
+
+        [TestMethod]
+        public void ShouldFailForRadiusLessThanZero()
+        {
+            Assert.ThrowsException<ArgumentException>(() =>
+            {
+                new LocationConstraintExtensionBuilder()
+                .WithRadius(-1)
+                .Build();
+            });
+        }
+
+        [TestMethod]
+        public void ShouldFailForUncertaintyLessThanZero()
+        {
+            Assert.ThrowsException<ArgumentException>(() =>
+            {
+                new LocationConstraintExtensionBuilder()
+                .WithMaxUncertainty(-1)
+                .Build();
+            });
+        }
+
+        [TestMethod]
+        public void ShouldBuildLocationConstraintWithGivenValues()
+        {
+            Extension<LocationConstraintContent> extension = new LocationConstraintExtensionBuilder()
+                .WithLatitude(_someLatitude)
+                .WithLongitude(_someLongitude)
+                .WithRadius(_someRadius)
+                .WithMaxUncertainty(_someUncertainty)
+                .Build();
+
+            Assert.AreEqual(Constants.Extension.LocationConstraint, extension.ExtensionType);
+            DeviceLocation deviceLocation = extension.Content.ExpectedDeviceLocation;
+            Assert.AreEqual(_someLatitude, deviceLocation.Latitude);
+            Assert.AreEqual(_someLongitude, deviceLocation.Longitude);
+            Assert.AreEqual(_someRadius, deviceLocation.Radius);
+            Assert.AreEqual(_someUncertainty, deviceLocation.MaxUncertainty);
+        }
+
+        [TestMethod]
+        public void ShouldBuildLocationConstraintWithDefaultValues()
+        {
+            Extension<LocationConstraintContent> extension = new LocationConstraintExtensionBuilder()
+                .WithLatitude(_someLatitude)
+                .WithLongitude(_someLongitude)
+                .Build();
+
+            Assert.AreEqual(Constants.Extension.LocationConstraint, extension.ExtensionType);
+            DeviceLocation deviceLocation = extension.Content.ExpectedDeviceLocation;
+            Assert.AreEqual(_someLatitude, deviceLocation.Latitude);
+            Assert.AreEqual(_someLongitude, deviceLocation.Longitude);
+            Assert.AreEqual(150d, deviceLocation.Radius);
+            Assert.AreEqual(150d, deviceLocation.MaxUncertainty);
```

Comment:
_(Same as other comments)_ Perhaps checking the serialised JSON output is more important than the getters?

Reply:
Added a test for this in 91595f3

Reply:
Added in 70924a6